### ISSUE TITLE
Gluon loses config when updating x86 devices from v2021.1.2 to v2023.1

### ISSUE THREAD

**Issue by lephisto** [freifunk-gluon/gluon#2967]
### Bug report

**What is the problem?**

We are updating from v2021.1.2 to v2023.1. Upgrade works 95% fine on Hardware routers/APs. On VMs and Futros the upgrade procedure reproduceably kills the Config. after reboot the VM/Offloader is in Config mode.

**What is the expected behaviour?**

I would expect the previous config to persist

**Gluon Version:**

v2021.1.2

**Site Configuration:**

https://github.com/ffrgb/site-ffrgb

**Custom patches:**

Nothing fancy.. 

**Screenshot - if that help:**

![gluon_upgrade](https://github.com/freifunk-gluon/gluon/assets/9072410/d4a4912c-68e4-4160-a360-b008f195168b)

**Comment by blocktrron** [comment]
@lephisto Do you know the disk size of said systems?

**Comment by lephisto** [comment]
In this case I just did a `qm importdisk` so the disk size went to 121MB. 

It happened already on a variety of disk sizes.

**Comment by maurerle** [comment]
I can confirm this issue with a vm upgraded from v2021.1.2 to v2023.1
Error message is the same, including the:

> Partition layout has changed. Full image will be written

Would be quite good to have a working upgrade path directly from the last openwrt-19 gluon release for us

**Comment by rotanid** [comment]
as there were no similar reports from v2021 to v2022, the issue has to be somewhat related to Gluon v2023.1 and not OpenWrt 22.03 ...

**Comment by neocturne** [comment]
The bug is in 2021.1.x, it fails to rescan the partition table when the layout has changed for some reason:
```
Commencing upgrade. Closing all shell sessions.
killall: telnetd: no process killed
Sending TERM to remaining processes ... ubusd askfirst urngd logd dnsmasq gluon-arp-limit netifd uradvd micrond sse-multiplexd uhttpd dnsmasq ntpd alfred batadv-vis respondd 
Sending KILL to remaining processes ... ntpd 
Switching to ramdisk...
[   37.438782] EXT4-fs (loop0): re-mounted. Opts: data=ordered
Performing system upgrade...
Reading partition table from bootdisk...
0+1 records in
0+1 records out
zcat: write error: Broken pipe
zcat: write: Broken pipe
Reading partition table from image...
30791+1 records in
30791+1 records out
partx: /dev/sda: error deleting partition 2
partx: /dev/sda: error adding partition 2
mount: mounting /dev/sda1 on /mnt failed: Invalid argument
umount: can't unmount /mnt: Invalid argument
Upgrade completed
Rebooting system...
```

partx should only fail like that when the partition is still busy, so maybe the umount of the rootfs is not working correctly...

**Comment by neocturne** [comment]
Updating from 2021.1.2 to 2022.1.1 to 2023.1 is working as expected, it only breaks when 2022.1 is skipped. As a first step, we should add a warning to the 2023.1 release notes.

I see two possible ways to proceed:
- Revert EFI boot support in 2023.1.x for now to avoid partition table change (or keep EFI as separate images)
- Accept that 2022.1.x is required as an intermediate step (I don't think it makes sense to put much effort into a fix on 2021.1.x, as an intermediate update from 2021.1.2 to 2023.1.x would be required anyways...)

**Comment by Djfe** [comment]
is it possible to remove efi boot support without breaking anything else? then I'm all for it.
move efi boot support to v2023.2.x

**Comment by Djfe** [comment]
I just informed the forum about this as a side note (in the changelog thread opened by @skorpy2009 )
Just to prevent further accidents.
I'll update the post Immediately once this issue got fixed properly

**Comment by neocturne** [comment]
Fixed with v2023.1.1.
